### ISSUE TITLE
Ability to zoom / open images fullscreen

### ISSUE THREAD

**Issue by goofballLogic** [YorgunKelebek/Peer-Programming-Blog#23]
With some of my screenshots, the default display size is too small to see the detail. It would be nice to have the ability to zoom into images and/or open them on their own fullscreen so I could view them using the browser's built-in image viewing capabilities.

**Comment by YorgunKelebek** [comment]
Yes, I agree this is needed. For now, browsers should already provide this:

![image](https://user-images.githubusercontent.com/56796626/74611054-86abca80-50f0-11ea-97fe-aec355656997.png)

**Comment by goofballLogic** [comment]
Yeah. We could also add very simply to:
```HTML
<img src="http://a.com/b.png" />
```
a wrapper for opening the image like this:
```HTML
<a href="http://a.com/b.png" target="_blank"><img src="http://a.com/b.png" ></a>
```

**Comment by YorgunKelebek** [comment]
Not tested thoroughly but this in the CSS works nicely at first glance:

```
.summary-body figure {
    transition: transform .2s; /* Animation */
    margin: 0 auto;
}

.summary-body figure:hover {
    transform: scale(1.5); /* (150% zoom - Note: if the zoom is too large, it will go outside of the viewport) */
}
```

**Comment by goofballLogic** [comment]
True but won't work on mobile which is where it's most needed.


-------- Original Message --------

**Comment by YorgunKelebek** [comment]
We could implement both, so on hover it would be a good visual aid for desktop users, and would open on click as well, while on mobile tap would open the image...

**Comment by goofballLogic** [comment]
Sounds good!

⁣Sent from BlueMail ​


-------- Original Message --------

**Comment by goofballLogic** [comment]
Hey I was just testing this, but the result is like this on my (small) laptop:

![Screen Shot 2020-02-23 at 15 23 30](https://user-images.githubusercontent.com/910448/75114737-8cedf980-5650-11ea-9fae-f48c2636fa37.png)

**Comment by goofballLogic** [comment]
Actually it gets worse because some of my blog posts have images at the top where the zoom makes images expand but get clipped by the non-expanded boundary of the blog item. We could maybe mitigate this (only this) problem by zooming only once a blog post has been expanded.

**Comment by YorgunKelebek** [comment]
OK, have a solution for this - will push it as an update tot he current PR, if OK?

```
@media only screen and (min-width: 1600px) {
    .summary-body:not(.blog-preview) figure:hover {
        transform: scale(1.5); /* (150% zoom - Note: if the zoom is too large, it will go outside of the viewport) */
    }
}
```
only zoom for large screens, and if `summary-body` doesn't have class `blog-preview`, which is when  expanded.

**Comment by goofballLogic** [comment]
worth a shot!

**Comment by goofballLogic** [comment]
This works really well now with my screen resolution. I vote to close!

A future refinement might be to allow a user to toggle this zooming on/off. This kind of UX tends to delight certain users and piss off others.

**Comment by YorgunKelebek** [comment]
Cool, cheers
